cmd/templ/generatecmd/main.go

```diff
@@ -203,6 +206,14 @@ func processChanges(ctx context.Context, fileNameToLastModTime map[string]time.T
 			return nil
 		}
 		if strings.HasSuffix(path, ".templ") {
+			pathSlices := regexp.MustCompile(`[\\/]`).Split(path, -1)
+			fileName := pathSlices[len(pathSlices)-1]
+			outputFileName := ""
+
+			if outputDir != "" {
```

Comment:
What should happen if the output directory doesn't exist? In the current behaviour, that can't happen, because it's operating on files that already exist in that directory.

I don't think this code accounts for templ files in subdirectories at the moment.

For example, if I've got a project layout of:

```
/shared/layouy/header/index.templ
/pages/default/index.templ
/pages/test.templ
```

I think I'd end up with one `index_templ.go` (i.e. I've lost one) and `test_templ.go` in the output directory.